jupyterhubutils/spawner/lsstspawner.py

```diff
@@ -0,0 +1,306 @@
+        # Run the superclass version, and then extract the fields
+        orig_pod = yield super().get_pod_manifest()
+        self.log.debug("Original pod: {}".format(orig_pod))
+        labels = orig_pod.metadata.labels.copy()
+        annotations = orig_pod.metadata.annotations.copy()
+        ctrs = orig_pod.spec.containers
+        cmd = None
+        if ctrs and len(ctrs) > 0:
+            cmd = ctrs[0].args or ctrs[0].command
+        # That should be it from the standard get_pod_manifest
+
+        # Get the standard env and then update it with the environment
+        # from our environment manager:
+        pod_env = self.get_env()
+        em = self.lsst_mgr.env_mgr
+        # The env_mgr's API tokens are stale; replace with current.
+        for envvar in ["JUPYTERHUB_API_TOKEN", "JPY_API_TOKEN"]:
+            em.set_env(envvar, None)
+            if envvar in pod_env:
+                em.set_env(envvar, pod_env[envvar])
+        em.refresh_pod_env()
+        pod_env.update(em.get_env())
+        self.log.debug("Initial pod env: %s" % json.dumps(pod_env,
+                                                          indent=4,
+                                                          sort_keys=True))
+        # If we do not have a UID for the user by now, we're sunk.  It
+        #  should have been set during authentication.
+        if not pod_env.get('EXTERNAL_UID'):
+            raise ValueError("Cannot determine user uid!")
+
+        # Now we do the custom LSST stuff
+
+        # Get image name
+        cfg = self.lsst_mgr.config
+        self.lab_service_account = None
+        if cfg.allow_dask_spawn:
+            self.lab_service_account = "dask"
+        pod_name = self.pod_name
+        image = (cfg.lab_default_image or
+                 self.image or
+                 self.orig_pod.image or
+                 "lsstsqre/sciplat-lab:latest")
+        tag = "latest"
+        size = None
+        image_size = None
+        # First pulls can be really slow for the LSST stack containers,
+        #  so let's give it a big timeout (this is in seconds)
+        self.http_timeout = 60 * 15
+        self.start_timeout = 60 * 15
+        # We are running the Lab at the far end, not the old Notebook
+        self.default_url = '/lab'
+        # We always want to check for refreshed images.
+        self.image_pull_policy = 'Always'
+        # Parse options form result.
+        clear_dotlocal = False
+        if self.user_options:
+            self.log.debug("user_options: " +
+                           json.dumps(self.user_options, sort_keys=True,
+                                      indent=4))
+            if self.user_options.get('kernel_image'):
+                image = self.user_options.get('kernel_image')
+                om = self.lsst_mgr.optionsform_mgr
+                size = self.user_options.get('size')
+                if size:
+                    image_size = om._sizemap[size]
```

Comment:
maybe make this a function to hide _sizemap, since it's not as private as you hope